### ISSUE TITLE
"Unsupported Confluence API call" when using versions >2.4.0

### ISSUE THREAD

**Issue by massdosage** [sphinx-contrib/confluencebuilder#991]
We use Sphinx and this extension to convert a number of `.md` and `.rst` files to HTML and then upload the files to Confluence. We have used this across a wide variety of projects with success for a number of years. We recently moved to the `2.5.0/1/2` versions and consistently get the below error which goes away if we downgrade to version `2.4.0`. Looking at the CHANGELOG my guess is there is some bug talking to one of the `v2` APIs. Happy to provide more information or run certain follow up steps if that would help you debug the root cause.

```
sphinxcontrib.confluencebuilder error:
---
Unsupported Confluence API call
An unsupported Confluence API call has been made. See the following
details for more information:
REQ: PUT
RSP: 400
URL: https://REDACTED.atlassian.net/wiki/
API: api/v2/pages/REDACTED_ID/properties
DATA: {
  "errors": [
    {
      "status": 400,
      "code": "INVALID_REQUEST_PARAMETER",
      "title": "Cannot update content property. Property : JsonContentProperty{id='JsonContentPropertyId{id=A_DIFFERENT_REDACTED_ID}', key='content-appearance-published', content=ExpandedReference{idProperties={id=ContentId{id=REDACTED_ID}, status=current, version=CollapsedReference{idProperties={number=198}, referentClass=class com.atlassian.confluence.api.model.content.Version}}, of=some(Content{id='ContentId{id=REDACTED_ID}', type=page, title='REDACTED PAGE TITLE', status=current, space=CollapsedReference{idProperties={key=DATA}, referentClass=class com.atlassian.confluence.api.model.content.Space}, history=CollapsedReference{idProperties={contentParent=CollapsedReference{idProperties={id=ContentId{id=REDACTED_ID}, status=current, version=CollapsedReference{idProperties={number=198}, referentClass=class com.atlassian.confluence.api.model.content.Version}}, referentClass=class com.atlassian.confluence.api.model.content.Content}}, referentClass=class com.atlassian.confluence.api.model.content.History}, version=CollapsedReference{idProperties={number=198}, referentClass=class com.atlassian.confluence.api.model.content.Version}, ancestors=null (CollapsedList), container=CollapsedReference{idProperties={key=DATA}, referentClass=class com.atlassian.confluence.api.model.content.Space}, childTypes=null (CollapsedMap)})}, version=Version{by=null, when=null, message='', number=2, syncRev='null', syncRevSource='null', ncsStepVersion='null', ncsStepVersionSource='null', confRev='null', content=EmptyReference{referentClass=class com.atlassian.confluence.api.model.content.Content}}} , isSuccessful: false, isAuthorized: true, isValid: false, hasErrors: true, Errors: \"jsonproperty.invalid.id\"",
      "detail": null
    }
  ]
}
```

**Comment by massdosage** [comment]
This looks similar to #974 but the status code, title and detail are different.

**Comment by jdknight** [comment]
To start off, a workaround to use a more recent version of this extension, you should be able to force original requests with other new features of this extension using:

```
confluence_api_mode = 'v1'
```

For sure though that it will not address the active issue mentioned. Having a 400 error being reported over a 500 one (i.e. https://github.com/sphinx-contrib/confluencebuilder/issues/974) is a bit more concerning. This extension might be performing calls in a way it should not. Just a matter of trying to reproduce the err scenario.

One step that might help is [generating a report](https://sphinxcontrib-confluencebuilder.readthedocs.io/en/stable/tips/#asking-for-help). This should help ensure development testing utilizes a similar configuration scenario.

Based on the provided output and inspecting the code, the issue appears to be specific to v2 (Cloud) [property update scenario](https://github.com/sphinx-contrib/confluencebuilder/blob/aac630f2263a7d4b1aa0f54ec1a4da17e1341327/sphinxcontrib/confluencebuilder/publisher.py#L1292-L1293). The implementation in this extension appears to following the [API call](https://developer.atlassian.com/cloud/confluence/rest/v2/api-group-content-properties/#api-pages-page-id-properties-property-id-put) -- maybe this extension is providing a bad key/prop-ID resulting in the 400 error.

If possible/willing, if you can try running with [`confluence_publish_debug`](https://sphinxcontrib-confluencebuilder.readthedocs.io/en/stable/configuration/#confval-confluence_publish_debug) configured to capture headers:

```
confluence_publish_debug = 'headers'
```

And just confirm on a failing `PUT` request that you see two unique IDs for the page and property (e.g. in the following example, `123456789` for the page and `987654321` for the property). This could rule out this extension forming a bad URL. Maybe checking if "Content-Length" is a non-zero.

<pre>
PUT https://example.atlassian.net/wiki/api/v2/pages/<b>123456789</b>/properties/<b>987654321</b>
User-Agent: Sphinx Confluence Builder
Accept-Encoding: gzip, deflate
Accept: application/json; charset=utf-8
Connection: keep-alive
X-Atlassian-Token: no-check
Content-Length: <b>291</b>
Content-Type: application/json
...
</pre>

This extension does not have a graceful way to dump the data sent in requests (maybe we should add that).

Another question to ask is if this is a consistent or a sporadically failure? Could help it point towards a possible concurrency issue versus just a bad request.

In the mean time, I will try to perform some publication requests that try to reproduce the content property update scenario and see if this is a consistent failure (that this extension failed to properly test) or try to trigger a scenario and check to see what workarounds exist.

Using `v1` API should be fine for users to fallback to now (if stuck). Ideally we want to improve v2 support as soon as we can before the cut-off date for v1 at the end of this year. I appreciate the reports and hopefully we can address this in a timely fashion.

**Comment by massdosage** [comment]
Thanks for the information, some answers:

* Yes, there are different IDs for page and property.
* This happens consistently every single time.

I will follow the steps you suggest to try and gather more information and provide it here when I get a chance (may only be next week).

Thanks for the prompt and detailed response!

**Comment by massdosage** [comment]
I ran the report as requested, here is the output from that:
```
------------[ cut here ]------------
(system)
 platform: Linux-5.13.0-1022-aws-x86_64-with-glibc2.36
   python: 3.11.9 (main, Jun 13 2024, 06:17:53) [GCC 12.2.0]
   sphinx: 7.3.7
 docutils: 0.20.1
 requests: 2.32.3
  builder: 2.4.0

(configuration)
confluence_cleanup_from_root: True
confluence_cleanup_purge: True
confluence_disable_notifications: True
confluence_ignore_titlefix_on_index: True
confluence_page_generation_notice: True
confluence_page_hierarchy: True
confluence_parent_page: (set)
confluence_publish: False
confluence_publish_dryrun: True
confluence_publish_prefix: REDACTED (It's of the form `word word -`)
confluence_server_pass: (set)
confluence_server_url: https://<removed> (cloud)
confluence_server_user: (set)
confluence_sourcelink: {'type': 'gitlab', 'host': 'gitlab.REDACTED.com', 'owner': 'REDACTED', 'repo': 'REDACTED', 'version': '0.1.0'}
confluence_space_key: (set; upper)
------------[ cut here ]------------
```

I also set `confluence_publish_debug = 'headers'` and this is the output from the failing call (there are 200 OKs before this from other pages):
```
(debug) Request]
PUT https://REDACTED.atlassian.net/wiki/api/v2/pages/REDACTED_83/properties/REDACTED_92
User-Agent: Sphinx Confluence Builder
Accept-Encoding: gzip, deflate
Accept: application/json; charset=utf-8
Connection: keep-alive
X-Atlassian-Token: no-check
Content-Length: 177
Content-Type: application/json
Authorization: Basic REDACTED_TOKEN=

(debug) Response]
Code: 400
Date: Mon, 01 Jul 2024 11:06:18 GMT
Content-Type: application/json;charset=utf-8
Server: AtlassianEdge
Atl-Confluence-Via: h:confluence-REDACTED.net
X-Content-Type-Options: nosniff
X-Xss-Protection: 1; mode=block
Atl-Traceid: REDACTED
Strict-Transport-Security: max-age=63072000; includeSubDomains; preload
Report-To: {"endpoints": [{"url": "https://REDACTED.cloudfront.net"}], "group": "endpoint-1", "include_subdomains": true, "max_age": 600}
Nel: {"failure_fraction": 0.001, "include_subdomains": true, "max_age": 600, "report_to": "endpoint-1"}
Vary: Accept-Encoding
Content-Encoding: gzip
Transfer-Encoding: chunked


sphinxcontrib.confluencebuilder error:

---
Unsupported Confluence API call

An unsupported Confluence API call has been made. See the following
details for more information:

REQ: PUT
RSP: 400
URL: https://REDACTED.atlassian.net/wiki/
API: api/v2/pages/REDACTED_83/properties
DATA: {
  "errors": [
    {
      "status": 400,
      "code": "INVALID_REQUEST_PARAMETER",
      "title": "Cannot update content property. Property : JsonContentProperty{id='JsonContentPropertyId{id=REDACTED_92}', key='content-appearance-published', content=ExpandedReference{idProperties={id=ContentId{id=REDACTED_83}, status=current, version=CollapsedReference{idProperties={number=213}, referentClass=class com.atlassian.confluence.api.model.content.Version}}, of=some(Content{id='ContentId{id=REDACTED_83}', type=page, title='REDACTED', status=current, space=CollapsedReference{idProperties={key=DATA}, referentClass=class com.atlassian.confluence.api.model.content.Space}, history=CollapsedReference{idProperties={contentParent=CollapsedReference{idProperties={id=ContentId{id=REDACTED_83}, status=current, version=CollapsedReference{idProperties={number=213}, referentClass=class com.atlassian.confluence.api.model.content.Version}}, referentClass=class com.atlassian.confluence.api.model.content.Content}}, referentClass=class com.atlassian.confluence.api.model.content.History}, version=CollapsedReference{idProperties={number=213}, referentClass=class com.atlassian.confluence.api.model.content.Version}, ancestors=null (CollapsedList), container=CollapsedReference{idProperties={key=DATA}, referentClass=class com.atlassian.confluence.api.model.content.Space}, childTypes=null (CollapsedMap)})}, version=Version{by=null, when=null, message='', number=2, syncRev='null', syncRevSource='null', ncsStepVersion='null', ncsStepVersionSource='null', confRev='null', content=EmptyReference{referentClass=class com.atlassian.confluence.api.model.content.Content}}} , isSuccessful: false, isAuthorized: true, isValid: false, hasErrors: true, Errors: \"jsonproperty.invalid.id\"",
      "detail": null
    }
  ]
}
---
```

**Comment by jdknight** [comment]
Thanks for the information so far. I tried reproducing a similar configuration set to see if it would help yield anything but no luck yet. I see in the report it does indicate an older builder (`builder: 2.4.0`), but I am going to assume that you may have generated a report on a local instance instance of a CI instance (which is fine; just mentioning just in case).

Looking at the outputs of the original and new responses look to show the issue is failing on a page's content property update. Specifically, trying to update a property `content-appearance-published`.

<details>
  <summary>Prettified 400 Error</summary>
<pre>
{
    "status": 400,
    "code": "INVALID_REQUEST_PARAMETER",
    "title": "Cannot update content property. Property : 
    JsonContentProperty{
        id='JsonContentPropertyId{id=REDACTED_92}',
        key='content-appearance-published',
        content=ExpandedReference{
            idProperties={
                id=ContentId{id=REDACTED_83},
                status=current,
                version=CollapsedReference{
                    idProperties={number=213},
                    referentClass=class com.atlassian.confluence.api.model.content.Version}
                },
                of=some(Content{
                    id='ContentId{id=REDACTED_83}',
                    type=page,
                    title='REDACTED',
                    status=current,
                    space=CollapsedReference{idProperties={key=DATA},
                    referentClass=class com.atlassian.confluence.api.model.content.Space
                },
                history=CollapsedReference{
                    idProperties={
                        contentParent=CollapsedReference{
                            idProperties={
                                id=ContentId{id=REDACTED_83},
                                status=current,
                                version=CollapsedReference{
                                    idProperties={number=213},
                                    referentClass=class com.atlassian.confluence.api.model.content.Version
                                }
                            },
                            referentClass=class com.atlassian.confluence.api.model.content.Content
                        }
                    },
                    referentClass=class com.atlassian.confluence.api.model.content.History
                },
                version=CollapsedReference{
                    idProperties={number=213},
                    referentClass=class com.atlassian.confluence.api.model.content.Version
                },
                ancestors=null (CollapsedList),
                container=CollapsedReference{
                    idProperties={key=DATA},
                    referentClass=class com.atlassian.confluence.api.model.content.Space
                },
                childTypes=null (CollapsedMap)
            })
        },
        version=Version{
            by=null,
            when=null,
            message='',
            number=2,
            syncRev='null',
            syncRevSource='null',
            ncsStepVersion='null',
            ncsStepVersionSource='null',
            confRev='null',
            content=EmptyReference{
                referentClass=class com.atlassian.confluence.api.model.content.Content
            }
        }
    },
    isSuccessful: false,
    isAuthorized: true,
    isValid: false,
    hasErrors: true,
    Errors: \"jsonproperty.invalid.id\"",
    "detail": null
}
</pre>
</details>

The details of this error show a similar structure to that of the PUT request. When updating the `content-appearance-published` property, a request like the following is made:

```
{
    'key': 'content-appearance-published',
    'value': 'default',
    'version': {
        'number': 5
    },
    'id': '123456789'
}
```

What I find interesting is that I do not see a hint related to the value configuration (e.g. `default`) in the 400 details output. It very well could be that a dump would not include this information, but hard to know having access to Confluence's code to check.

A current suspicion is that maybe the extension is re-pushing the `content-appearance-publish` property that it originally fetched back to Confluence (ideally, should not happen if there is "no change"), and that the value of `content-appearance-publish` now is either in an odd state or cannot be reused between the GET and PUT events.

Would you mind trying to open the following URL for the site and page where the error occurs?

```
https://YOUR_SITE.atlassian.net/wiki/api/v2/pages/YOUR_PAGE_ID/properties?key=content-appearance-published
```

You should see an output like this:

```
{
  "results": [
    {
      "value": "default",
      "key": "content-appearance-published",
      "id": "123456789",
      "version": {
        "number": 5,
        "message": "",
        "minorEdit": false,
        "authorId": "123456789:<guid>",
        "createdAt": "2024-07-01T15:30:11.904Z"
      }
    }
  ],
  "_links": {
    "base": "https://YOUR_SITE.atlassian.net/wiki"
  }
}
```

And I'm curious to know if the output of `value` is something not expected (most likely will be sane, but want to rule this out).


----

The only other suggestion I have at this time is maybe try running with the [development installation](https://sphinxcontrib-confluencebuilder.readthedocs.io/en/stable/install/#development-installation) and try to set the following option:

```
confluence_publish_debug = 'headers_and_data'
```

This should output the contents of the PUT request made which might real if this extension is sending a malformed request out.

**Comment by massdosage** [comment]
This is all running on a CI instance.

I tried hitting the URL `https://YOUR_SITE.atlassian.net/wiki/api/v2/pages/YOUR_PAGE_ID/properties?key=content-appearance-published` directly as suggested and it returned output pretty much exactly like what you said above. `value` was set to `default`. 

Then, instead of installing `sphinxcontrib-confluencebuilder==2.5.2` I did `pip install git+https://github.com/sphinx-contrib/confluencebuilder.git` to get the latest version and then the build succeeded. So I guess whatever the issue is, it's been fixed by whatever has been changed between `2.5.2` and what is now in `main`. So this will probably work in the next release which is good news!

**Comment by jdknight** [comment]
Gah ~ well.... glad the development revision works. Maybe it was the https://github.com/sphinx-contrib/confluencebuilder/commit/8bd8137178d65bd35040020fdaa1fb724aa8e2a0 commit? Sorry for the issues.

Will flag as "available-next-release" with the assumption that the now working state was related to a fix in the development branch. We should be building a release soon (maybe this or next weekend; no guarantees yet) to wrap this up. In the interim, if you do see any other issues, please let me know.

**Comment by massdosage** [comment]
No worries, thanks for looking into this and all the prompt responses. If you want to ping me here when the release comes out I can then try it out and report back if its fixed. No rush on our side as we'll just stick with `2.4.0` for now.

**Comment by jdknight** [comment]
v2.6 is now available on PyPI -- marking as closed.

It is recommended to try the new release without any explicit v1 API mode and please report any issues. Atlassian will drop support for v1 by the end of this year, and we hope to make sure this extension is in a good state before that time. Thanks again for the initial report.

**Comment by massdosage** [comment]
Thanks, I can confirm that `2.6.0` works for the issue reported here. I also tried `2.5.2` again to reproduce the error and bizarrely this no longer fails. I wonder if Atlassian fixed something on their side too and the cloud version has been upgraded in the last week or two. Either way this new version looks good and we'll start moving over to it. Thanks!

**Comment by hasier** [comment]
Hi! I wasn't sure whether to open a new issue or follow up on this one. I've been following this for a while as we were having the exact same problem, and even after bumping to 2.6.0+ it still consistently fails for us, so it felt like a follow up here would make sense. Please let me know if you'd rather have a new issue opened and I'll do so.

We have experienced everything **massdosage** went through exactly as mentioned, except for the fact that it is not resolved for us 😅 We are currently pinned at `2.4.0`, which is fine for now, but it'd be good to be able to move to more recent versions. Please see below the debugging logs as requested previously in this thread.

We have not tried using `v1` as [I saw in this thread it is discouraged](https://github.com/sphinx-contrib/confluencebuilder/issues/991#issuecomment-2192858363).

<details>
  <summary>Request to fetch property</summary>

```
GET https://YOUR_SITE.atlassian.net/wiki/api/v2/pages/YOUR_PAGE_ID/properties?key=content-appearance-published

{
  "results": [
      {
        "version": {
        "number": 1,
        "message": "",
        "minorEdit": false,
        "authorId": "",
        "createdAt": "2022-07-26T06:46:57.433Z"
      },
      "value": "full-width",
      "key": "content-appearance-published",
      "id": REDACTED
    }
  ],
  "_links": {
    "base": "https://REDACTED.atlassian.net/wiki"
  }
}
```
</details>

<details>
  <summary>Report</summary>
<br/>

Running in a `ubuntu-latest` Github Action.

```
(system)
 platform: Linux-6.8.0-1017-azure-x86_64-with-glibc2.39
   python: 3.11.11 (main, Dec 4 2024, 12:58:35) [GCC 13.2.0]
   sphinx: 8.1.3
 docutils: 0.21.2
 requests: 2.32.3
  urllib3: 2.3.0
  builder: 2.9.0
(configuration)
confluence_ask_password: False
confluence_page_generation_notice: True
confluence_page_hierarchy: True
confluence_parent_page: (set)
confluence_publish: True
confluence_publish_debug: headers_and_data
confluence_server_pass: (set)
confluence_server_url: https://(removed) (cloud)
confluence_server_user: (set)
confluence_space_key: (set; upper)
(confluence instance)
 connected: yes
   fetched: yes
   decoded: yes
    parsed: yes
      type: confluence
   version: 1000.0.0-43d301f101ec
     build: 6452
```
</details>

<details>
  <summary>Last 2 requests</summary>
<br/>

Using `confluence_publish_debug = "headers_and_data"`
  
 ```
(debug) Request: get_page_property]
GET https://REDACTED.atlassian.net/wiki/api/v2/pages/REDACTED/properties?key=content-appearance-published
User-Agent: SphinxConfluenceBuilder/2.9.0
Accept-Encoding: gzip, deflate, br
Accept: application/json; charset=utf-8
Connection: keep-alive
X-Atlassian-Token: no-check
Authorization: (redacted)
(debug) Response]
Code: 200
Date: Wed, 22 Jan 2025 10:08:49 GMT
Content-Type: application/json;charset=utf-8
Server: AtlassianEdge
Vary: Accept-Encoding
X-Confluence-Request-Time: 1737540528932
Link: <https://REDACTED.atlassian.net/wiki>; rel="base"
Content-Language: en-US
Content-Encoding: gzip
Atl-Confluence-Via: REDACTED
X-Content-Type-Options: nosniff
X-Xss-Protection: 1; mode=block
Atl-Traceid: REDACTED
Atl-Request-Id: REDACTED
Strict-Transport-Security: max-age=63072000; includeSubDomains; preload
Report-To: {"endpoints": [{"url": "https://REDACTED.cloudfront.net"}], "group": "endpoint-1", "include_subdomains": true, "max_age": 600}
Nel: {"failure_fraction": 0.001, "include_subdomains": true, "max_age": 600, "report_to": "endpoint-1"}
Server-Timing: atl-edge;dur=167,atl-edge-internal;dur=14,atl-edge-upstream;dur=154,atl-edge-pop;desc="aws-us-east-1"
Transfer-Encoding: chunked
(debug) Response data]
{
  "results": [
    {
      "version": {
        "number": 1,
        "message": "",
        "minorEdit": false,
        "authorId": "",
        "createdAt": "2022-07-26T06:46:57.433Z"
      },
      "value": "full-width",
      "key": "content-appearance-published",
      "id": "REDACTED"
    }
  ],
  "_links": {
    "base": "https://REDACTED.atlassian.net/wiki"
  }
}
(debug) Request: store_page_property]
PUT https://REDACTED.atlassian.net/wiki/api/v2/pages/REDACTED/properties/REDACTED
User-Agent: SphinxConfluenceBuilder/2.9.0
Accept-Encoding: gzip, deflate, br
Accept: application/json; charset=utf-8
Connection: keep-alive
X-Atlassian-Token: no-check
Content-Length: 177
Content-Type: application/json
Authorization: (redacted)
(debug) Request data]
(non-json)
(debug) Response]
Code: 400
Date: Wed, 22 Jan 2025 10:08:49 GMT
Content-Type: application/json;charset=utf-8
Server: AtlassianEdge
X-Confluence-Request-Time: 1737540529119
Content-Language: en-US
Atl-Confluence-Via: REDACTED
X-Content-Type-Options: nosniff
X-Xss-Protection: 1; mode=block
Atl-Traceid: REDACTED
Atl-Request-Id: REDACTED
Strict-Transport-Security: max-age=63072000; includeSubDomains; preload
Report-To: {"endpoints": [{"url": "https://REDACTED.cloudfront.net"}], "group": "endpoint-1", "include_subdomains": true, "max_age": 600}
Nel: {"failure_fraction": 0.001, "include_subdomains": true, "max_age": 600, "report_to": "endpoint-1"}
Content-Encoding: br
Vary: Accept-Encoding
Server-Timing: atl-edge;dur=176,atl-edge-internal;dur=13,atl-edge-upstream;dur=163,atl-edge-pop;desc="aws-us-east-1"
Transfer-Encoding: chunked
(debug) Response data]
{
  "errors": [
    {
      "status": 400,
      "code": "BAD_REQUEST",
      "title": "Cannot update content property. Property : JsonContentProperty{id='JsonContentPropertyId{id=REDACTED}', key='content-appearance-published', content=ExpandedReference{idProperties={version=CollapsedReference{idProperties={number=297}, referentClass=class com.atlassian.confluence.api.model.content.Version}, status=current, id=ContentId{id=REDACTED}}, of=some(Content{id='ContentId{id=REDACTED}', type=page, title='Dictionary Objects', status=current, space=CollapsedReference{idProperties={key=CON}, referentClass=class com.atlassian.confluence.api.model.content.Space}, history=CollapsedReference{idProperties={contentParent=CollapsedReference{idProperties={version=CollapsedReference{idProperties={number=297}, referentClass=class com.atlassian.confluence.api.model.content.Version}, status=current, id=ContentId{id=REDACTED}}, referentClass=class com.atlassian.confluence.api.model.content.Content}}, referentClass=class com.atlassian.confluence.api.model.content.History}, version=CollapsedReference{
      "detail": null
    }
  ]
}
sphinxcontrib.confluencebuilder error:
---
Unsupported Confluence API call
An unsupported Confluence API call has been made. See the following
details for more information:
REQ: PUT
RSP: 400
URL: https://REDACTED.atlassian.net/wiki/
API: api/v2/pages/REDACTED/properties
DATA: {
  "errors": [
    {
      "status": 400,
      "code": "BAD_REQUEST",
      "title": "Cannot update content property. Property : JsonContentProperty{id='JsonContentPropertyId{id=REDACTED}', key='content-appearance-published', content=ExpandedReference{idProperties={version=CollapsedReference{idProperties={number=297}, referentClass=class com.atlassian.confluence.api.model.content.Version}, status=current, id=ContentId{id=REDACTED}}, of=some(Content{id='ContentId{id=REDACTED}', type=page, title='Dictionary Objects', status=current, space=CollapsedReference{idProperties={key=CON}, referentClass=class com.atlassian.confluence.api.model.content.Space}, history=CollapsedReference{idProperties={contentParent=CollapsedReference{idProperties={version=CollapsedReference{idProperties={number=297}, referentClass=class com.atlassian.confluence.api.model.content.Version}, status=current, id=ContentId{id=REDACTED}}, referentClass=class com.atlassian.confluence.api.model.content.Content}}, referentClass=class com.atlassian.confluence.api.model.content.History}, version=CollapsedReference{
      "detail": null
    }
  ]
}
```
</details>
